### ISSUE TITLE
help message not clear

### ISSUE THREAD

**Issue by Masber** [rust-lang/rust-clippy#10217]
### Summary

I got the following clippy warning which help message is missleading

```
warning: called `unwrap` on `first` after checking its variant with `is_some`
   --> src/shasta/cfs/session.rs:237:47
    |
235 |             if first.is_some() {
    |             ------------------ help: try: `if let Some(..) = first`
236 |                 
237 |                 ansible_limits = String::from(first.unwrap());
    |                                               ^^^^^^^^^^^^^^
    |
    = note: `#[warn(clippy::unnecessary_unwrap)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_unwrap
```

The problem is here

> try: `if let Some(..) = first`

I don't understand why the `..` when in reality it is asking for a var name, to me `..` means a slice. I think this is confusing and a placeholder should be instead like:

> try: `if let Some(<item>) = first` 

Other Clippy help messages uses place holders like the one below:

```
help: consider using an iterator
    |
217 |                 for (i, <item>) in target_groups_json.iter().enumerate().skip(1) {
    |                     ~~~~~~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Reproducer

I tried this code:

```rust
            if first.is_some() {
                
                ansible_limits = String::from(first.unwrap());
```

I expected to see this happen:
```
warning: called `unwrap` on `first` after checking its variant with `is_some`
   --> src/shasta/cfs/session.rs:237:47
    |
235 |             if first.is_some() {
    |             ------------------ help: try: `if let Some(<item>) = first`
236 |                 
237 |                 ansible_limits = String::from(first.unwrap());
    |                                               ^^^^^^^^^^^^^^
    |
    = note: `#[warn(clippy::unnecessary_unwrap)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_unwrap
```
Instead, this happened:

```
warning: called `unwrap` on `first` after checking its variant with `is_some`
   --> src/shasta/cfs/session.rs:237:47
    |
235 |             if first.is_some() {
    |             ------------------ help: try: `if let Some(..) = first`
236 |                 
237 |                 ansible_limits = String::from(first.unwrap());
    |                                               ^^^^^^^^^^^^^^
    |
    = note: `#[warn(clippy::unnecessary_unwrap)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_unwrap
```

### Version

```text
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: x86_64-unknown-linux-gnu
release: 1.64.0
LLVM version: 14.0.6
```
